pages/writing.md

```diff
@@ -45,23 +45,131 @@ juliaup status
 * [emacs](https://www.gnu.org/software/emacs/) / [vim](https://www.vim.org/) / other IDEs + [JuliaEditorSupport](https://github.com/JuliaEditorSupport)
 * [Jupyter](https://jupyter.org/) / [IJulia.jl](https://github.com/JuliaLang/IJulia.jl)
 * [Pluto.jl](https://plutojl.org/)
 
 ## Running code
 
-* [REPL](https://docs.julialang.org/en/v1/stdlib/REPL/)
-* [Revise.jl](https://github.com/timholy/Revise.jl)
-* [running in VSCode](https://www.julia-vscode.org/docs/stable/userguide/runningcode/)
-* startup file
+All executed Julia code ends up in a REPL[^1], but how it gets there can vary greatly.
+
+[^1]: Unless you're compiling binaries with [StaticTools.jl](https://github.com/brenhinkeller/StaticTools.jl).
+
+<!-- The VSCode is important to write in tandem with or after the Development environments part is written to avoid overlap. -->
+Most people develop Julia using a tool that allows for [interactive development](https://en.wikipedia.org/wiki/Interactive_programming).
+In VScode, you can [run code](https://www.julia-vscode.org/docs/stable/userguide/runningcode/) in the following useful ways:
+* For interactive development, the standard way to use Julia, you can use hotkeys to run selections, lines, sections, or whole files in the currently open REPL.
+    * Select a part of code to run only that part,
+    * Put your cursor inside or after a language construct, such as a definition or function call, to run the whole construct,
+    * Use the "run file in REPL" hotkey to do just that.
+* You can run entire scripts from a new Julia session each time, but this isn't great as it takes a while to start a new REPL each time.
+    To work with a (more or less) fresh workspace each time, you should bundle code into [local packages](#package-management).
+
+Some code, however, you want to run on startup, every time Julia is loaded.
+While VSCode has settings that allow you to specify startup flags, more powerful is its ability to automatically run a [startup file](https://docs.julialang.org/en/v1/manual/command-line-interface/#Startup-file).
+This allows you to, among other things, load packages that improves your development experience or contain functionality that you use very often.
+
+It's tempting to put a lot of code in your `startup.jl` file for convenience's sake, but this convenience comes at a cost of increasing your startup time as it is executed.
+To balance convenience and time, we can use AST transformations to conditionally load functionality from packages only when necessary.
+We tell the REPL to listen for a set of user-defined functions or macros every time a command is sent to be executed;
+when one is detected, before executing the command, it will first execute the corresponding `using` statement to allow the function or macro to be called.
+
+```julia
+ENV["JULIA_EDITOR"] = "code"
+
+using Revise
+using OhMyREPL
```

Comment:
Revise and OhMyREPL both recommend loading the packages in a `try-catch` statement:

https://timholy.github.io/Revise.jl/stable/config/#Using-Revise-by-default-1 
```Julia
try
    using Revise
catch e
    @warn "Error initializing Revise" exception=(e, catch_backtrace())
end
```

https://kristofferc.github.io/OhMyREPL.jl/latest/installation/#Automatically-start-with-Julia.
```Julia
atreplinit() do repl
    try
        @eval using OhMyREPL
    catch e
        @warn "error while importing OhMyREPL" e
    end
end
```